core/src/main/java/hudson/Util.java

```diff
@@ -1180,14 +1177,16 @@ public static void touch(@Nonnull File file) throws IOException {
 
     /**
      * Copies a single file by using Ant.
+     *
+     * @deprecated use {@link Files#copy(Path, Path, CopyOption...)}
      */
+    @Deprecated
     public static void copyFile(@Nonnull File src, @Nonnull File dst) throws BuildException {
-        Copy cp = new Copy();
-        cp.setProject(new org.apache.tools.ant.Project());
-        cp.setTofile(dst);
-        cp.setFile(src);
-        cp.setOverwrite(true);
-        cp.execute();
+        try {
+            Files.copy(src.toPath(), dst.toPath(), StandardCopyOption.COPY_ATTRIBUTES, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Since `File#toPath` can throw `InvalidPathException`, we created the `fileToPath` method in this class which wraps `InvalidPathException` in `IOException` so that existing callers who handle `IOException` don't need to be updated to handle `InvalidPathException` as well. You should use that instead of calling `toPath` directly.